batch/batch/batch.py

```diff
@@ -1067,7 +1067,9 @@ def is_successful(self):
 
 
 async def update_job_with_pod(job, pod):
-    if pod and (job.is_complete() or pod.metadata.labels['task'] != job._current_task.name):
+    if pod and (job.is_complete() or
+                job._current_task.name is None or
```

Comment:
The problem is that if the job is not complete than the current task should not be None. So we can merge this, but it's not solving the problem